mapbox/app/src/main/java/com/mapbox/services/android/testapp/geocoding/GeocodingServiceActivity.java

```diff
@@ -161,29 +169,42 @@ private void getLastLocation() {
       }
 
       // It is possible that the user presses the button to get the address before the
-      // LostApiClient object successfully connects. In such a case, addressRequested
+      // location engine object successfully connects. In such a case, addressRequested
       // is set to true, but no attempt is made to fetch the address (see
       // fetchAddressButtonHandler()) . Instead, we start the intent service here if the
-      // user has requested an address, since we now have a connection to LostApiClient.
+      // user has requested an address, since we now have a connection to location engine.
       if (addressRequested) {
         startIntentService();
       }
     }
   }
 
   @Override
-  protected void onStop() {
-    super.onStop();
-    if (lostApiClient.isConnected()) {
-      lostApiClient.disconnect();
+  public void onConnected() {
+    getLastLocation();
+  }
+
+  @Override
+  public void onLocationChanged(Location location) {
+    if (location != null) {
+      getLastLocation();
```

Comment:
Actually, we're not requesting location updates, so it's not necessary.
Problem. Are we going to leave `onLocationChanged` method empty? Should we segregate `LocationEngineListener` interface so client is not forced to depend on methods it does not use? What do you think? @zugaldia